### PR TITLE
Use gmres only with OMP

### PR DESCRIPTION
**Pull request by blegouix**
After Gysela meeting discussion, we keep bicgstab on GPU

### CODE REVIEW

File: include/ddc/kernels/splines/matrix_sparse.hpp [hunk 1]
```diff
@@ -98,13 +98,22 @@ template <class ExecSpace>
 class Matrix_Sparse : public Matrix
 {
     using matrix_sparse_type = gko::matrix::Csr<double, int>;
+#ifdef KOKKOS_ENABLE_OPENMP
+    using solver_type = std::conditional_t<
+            std::is_same_v<ExecSpace, Kokkos::OpenMP>,
+            gko::solver::Gmres<double>,
+            gko::solver::Bicgstab<double>>;
+#else
+    using solver_type = std::shared_ptr<gko::solver::Gmres<double>>;
```
Comment (blegouix) [inline include/ddc/kernels/splines/matrix_sparse.hpp:107]:
```suggestion
    using solver_type = std::shared_ptr<gko::solver::Bicgstab<double>>;
```

**Comment by blegouix** [comment]
If we can quickly merge it then I propagate in gysela to fix the CI

File: include/ddc/kernels/splines/matrix_sparse.hpp [hunk 2]
```diff
@@ -98,13 +98,22 @@ template <class ExecSpace>
...
+#endif
+
 
 private:
     std::unique_ptr<gko::matrix::Dense<double>> m_matrix_dense;
 
     std::shared_ptr<matrix_sparse_type> m_matrix_sparse;
 
-    std::shared_ptr<gko::solver::Gmres<double>> m_solver;
+    std::shared_ptr<solver_type> m_solver;
```
Comment (tpadioleau) [inline include/ddc/kernels/splines/matrix_sparse.hpp:116]:
```suggestion
    std::shared_ptr<gko::LinOp> m_solver;
```

Will be even simpler to change the solver

**Review by tpadioleau — APPROVED** [review]
Yes good to me

File: include/ddc/kernels/splines/matrix_sparse.hpp (hunk 2 above)
Comment (blegouix) [reply include/ddc/kernels/splines/matrix_sparse.hpp:116]: This does not compile:

image

Comment (tpadioleau) [reply include/ddc/kernels/splines/matrix_sparse.hpp:116]:
```
        std::shared_ptr const solver = solver_factory->generate(m_matrix_sparse);
        m_solver = solver;
        m_solver_tr = solver->transpose();
```
In fact I think could be done like this.

Comment (tpadioleau) [reply include/ddc/kernels/splines/matrix_sparse.hpp:116]: It's ok for now

**Pull request merged**